### DISCUSSION TITLE
Generate schema.ts from existing frontend components

### DISCUSSION THREAD

**Discussion by Perni1984** [tinacms/tinacms#2177] [Ideas]
We are having an integrated workflow were we are building our design system components according to atomic design in figma. Our use case is that basically all our organisms are the building blocks that we would like to make TinaCMS enabled. We have a half-automated way of generating our react components out of figma aswell as their prop typescript types. Currently we are exploring to parsing the typescript types of those components automatically and generating the schema.ts file out of it. Currently we are playing around with https://github.com/vega/ts-json-schema-generator to do that - but we are not yet sure if this is the right approach.

What do you guys think? Is this in aligment with your product vision?

**Comment by jeffsee55** [comment]
Our schema is designed to be "compatible" with JSON schema, having said that we haven't done a ton to work on that tooling. 

We've discussed having some tools similar to what you're doing, but that we were thinking it'd look at content (ie. `.md` or `.json` files) and build out schemas based on those (likely utilizing JSON schema tooling). I can definitely see a use case for your idea too (I'd love to use that)

Basically, the bulk of work we _need_ and want to do is JSON schemas to Tina Schemas. And that'd really open things up to what you're after. Let's keep this thread open and we can continue to discuss how this might look. Thanks for opening up the discussion!
